### DISCUSSION TITLE
查询数据时报错 {"E":"invalid type: map, expected i32"}

### DISCUSSION THREAD

**Discussion by lopo1** [rbatis/rbatis#508] [Q&A]
使用版本：
rbs = { version = "4.5"}
rbatis = { version = "4.5",features = []}
rbdc-mysql= { version="4.5" }
reqwest = { version = "0.12", features = ["json"] }
impl_select!(BlockScan{select_by_tag(tag:&str) -> Option => "`where tag = #{tag} limit 1`"});
let list = BlockScan::select_by_tag(&mut GLOBAL_DB.clone(), tag).await;
 println!("select_by_id = {}", json!(list));

#[derive(Clone, Debug, Serialize, Deserialize,Default)]
#[serde(default)]
pub struct BlockScan {
    pub id: Option<i32>,
    pub tag: Option<String>,
    pub block_number: Option<i64>,
}
报错如下：
select_by_id = {"Err":{"E":"invalid type: map, expected i32"}}

**Comment by lopo1** [comment]
![image](https://github.com/rbatis/rbatis/assets/22266702/829e2701-f696-44da-85aa-1c0a0b32cd5a)
MySQL查询接口

**Comment by lopo1** [comment]
改成Vec就不会报错，impl_select!(BlockScan{select_by_tag(tag:&str) -> Vec => "`where tag = #{tag} limit 1`"}); 但是我不想使用数组，只想得到一条数据结果

**Reply by zhuxiujia** [reply]
你运行下carogo update更新下依赖库，然后开启debug_mode看看哪个字段不对

**Reply by zhuxiujia** [reply]
你运行下carogo update更新下依赖库，然后开启debug_mode看看哪个字段不对

**Reply by lopo1** [reply]
![image](https://github.com/rbatis/rbatis/assets/22266702/bd2fa9a1-52d7-44bf-8ccc-c3a8f6e57f02)
输出的是yg一个数组，有点奇怪

**Reply by zhuxiujia** [reply]
如果是最新版本的RBatis，会提示`pub id: Option,
pub tag: Option,
pub block_number: Option,` 其中一个字段哪个字段类型不匹配的
